### DISCUSSION TITLE
"is Foo" works but "is not Foo" doesn't when there is a Type Foo and other member with the same name Foo

### DISCUSSION THREAD

**Discussion by Liero** [dotnet/csharplang#4743] [General]
I'm not sure whether this is a compiler bug or a language feature, but:

```csharp
class Foo {}
public class Program
{
	object Foo {get;}
	
	public static void Main()
	{
		var program = new Program();
		if (program.Foo is Foo) 
		{
			Console.WriteLine("Foo is of type Foo");
		}
		if (program.Foo is not Foo)  //Compilation error: An object reference is required for the non-static field, method, or property 'Program.Foo'
		{
		    Console.WriteLine("Foo is not of type Foo");
		}
		
	}
}
```

see https://dotnetfiddle.net/KrmpAk

**Comment by HaloFour** [comment]
See:  https://github.com/dotnet/roslyn/issues/50671

TL;DR: the behavior is by design, but it sounds like a few members of the LDM were willing to talk about it further and consider a breaking change.  Despite the last comment on that issue it doesn't seem like there was a followup on this repo.
